### PR TITLE
BUG: Add SparseArray.all

### PR DESCRIPTION
**Pull request by Licht-T**
- [x] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

This is the part of https://github.com/pandas-dev/pandas/pull/17386.
`Block.where` uses `ndarray.all`, but there is no such method in `SparseArray`.
This makes that `all` evaluates `SparseArray([False, False, True])` as `True`.
https://github.com/pandas-dev/pandas/blob/master/pandas/core/internals.py#L1400

### CODE REVIEW

File: pandas/core/sparse/array.py [hunk 1]
```diff
@@ -614,6 +614,24 @@ def fillna(self, value, downcast=None):
...
+        Returns
+        -------
+        all : bool
+        """
+        nv.validate_all(args, kwargs)
+
+        values = self.sp_values
+
+        if len(values) != len(self):
+            values = values.tolist()
```
Comment (jreback) [inline pandas/core/sparse/array.py]: is there a reason you are converting to a list? and not a dense array?

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatsnew note. do we have a specific issue for this? (other that the PR)?

File: pandas/core/sparse/array.py [hunk 2]
```diff
@@ -614,6 +614,24 @@ def fillna(self, value, downcast=None):
         return self._simple_new(new_values, self.sp_index,
                                 fill_value=fill_value)
 
+    def all(self, axis=0, *args, **kwargs):
+        """
```
Comment (jreback) [inline pandas/core/sparse/array.py]: might as well add any as well.

File: pandas/core/sparse/array.py (hunk 1 above)
Comment (Licht-T) [reply pandas/core/sparse/array.py]: @jreback For the performance reason. This is sparse and the converting to dense is not efficient.

Comment (jreback) [reply pandas/core/sparse/array.py]: I am asking why you are converting ``.tolist()``, then appending, that is non-performant.

Comment (Licht-T) [reply pandas/core/sparse/array.py]: Simply I wanted to use `np.all`. I got an idea not to use `tolist()`. I'll change the solution.

**Comment by Licht-T** [comment]
@jreback Thanks for your review.
- Changed the `fill_value` check and now we are not using `tolist()`
- Also added `SparseArray.any`.
- There seems to be no specific issue for this.

File: doc/source/whatsnew/v0.21.0.txt [hunk 3]
```diff
@@ -544,6 +544,7 @@ Sparse
 
 - Bug in ``SparseSeries`` raises ``AttributeError`` when a dictionary is passed in as data (:issue:`16905`)
 - Bug in :func:`SparseDataFrame.fillna` not filling all NaNs when frame was instantiated from SciPy sparse matrix (:issue:`16112`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: say that these are now implemented to handle SparseArray

File: pandas/core/sparse/array.py [hunk 4]
```diff
@@ -614,6 +614,40 @@ def fillna(self, value, downcast=None):
...
+        Tests whether all elements evaluate True
+
+        Returns
+        -------
+        all : bool
```
Comment (jreback) [inline pandas/core/sparse/array.py]: add a See Also to point to np.all

**Review by jreback — APPROVED** [review]
minor comment. ping on green

**Comment by Licht-T** [comment]
@jreback Thank you. These are now fixed.

**Comment by Licht-T** [comment]
@jreback I found the bug in this solution. When `fill_value = np.array([1,1,1])`, `np.all(fill_value)` returns True. I'll fix.

**Comment by jreback** [comment]
lgtm. waiting for CI to go green again before merging (unrelated conda-ish issues)

**Comment by jreback** [comment]
can you rebase

**Comment by Licht-T** [comment]
@jreback Rebased.

**Comment by jreback** [comment]
thanks @Licht-T

**Pull request merged**